fmriprep/workflows/bold/base.py

```diff
@@ -561,28 +560,31 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
         ])
 
         if t2s_coreg:
-            # create a T2* map
-            bold_t2s_wf = init_bold_t2s_wf(bold_echos=bold_file,
-                                           echo_times=tes,
-                                           mem_gb=mem_gb['resampled'],
-                                           omp_nthreads=omp_nthreads,
-                                           name='bold_t2s_wf')
-            bold_t2s_wf.inputs.inputnode.name_source = ref_file
-
             # Replace EPI-to-T1w registration inputs
             workflow.disconnect([
                 (bold_sdc_wf, bold_reg_wf, [
                     ('outputnode.bold_ref_brain', 'inputnode.ref_bold_brain'),
                     ('outputnode.bold_mask', 'inputnode.ref_bold_mask')]),
             ])
             workflow.connect([
-                (bold_hmc_wf, bold_t2s_wf, [
-                    ('outputnode.xforms', 'inputnode.hmc_xforms')]),
-                (bold_t2s_wf, bold_reg_wf, [
+                (bold_bold_trans_wf, bold_reg_wf, [
                     ('outputnode.t2s_map', 'inputnode.ref_bold_brain'),
-                    ('outputnode.oc_mask', 'inputnode.ref_bold_mask')]),
+                    ('outputnode.t2s_mask', 'inputnode.ref_bold_mask')]),
             ])
 
+    workflow.connect([
+        # Connect bold_bold_trans_wf -- either original or T2S shadow
+        (inputnode, bold_bold_trans_wf, [
+            ((if multiecho 'ref_file' else 'bold_file'), 'inputnode.name_source')]),
```

Comment:
I'm a little stuck here. I seem to remember seeing something like this at one point (maybe @effigies or @oesteban had done something similar ?), but I can't figure out a nipype appropriate way to do this. Any pointers would be appreciated !

Reply:
This is a ternary operator with a minor syntax error. Try `'ref_file' if multiecho else 'bold_file'`.

Reply:
That was it ! Sorry, I had it set up like an if-statement. Thanks !